pycoal/mineral.py

```diff
@@ -303,9 +304,35 @@ def __init__(self, algorithm=SAM, **kwargs):
             **kwargs: arguments that will be passed to the chosen classifier
         """
 
-        # set the user's chosen classifier 
-        self.algorithm = algorithm
-
+        # parse config file
+        config = configparser.ConfigParser()
+        config.read('config.ini')
+
+        # use the user's chosen classifier
+        set_algo = config['processing']['algo']
+        
+        # use the user's chosen implementation
+        set_impl = config['processing']['impl']
+        
+        if set_algo == 'SAM':
+                if set_impl == 'serial':
+                        self.algorithm = SAM
+                        '''
```

Comment:
Never leave code which is not being used. Please uncomment this logic.

Reply:
Also, eventually, I want the default to by pytorch. Simply because it is most performant.